### PR TITLE
[api-draft][catalog] jdbc catalog

### PR DESCRIPTION
**Pull request by ashulin**
## Purpose of this pull request
add jdbc catalog

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -514,6 +515,13 @@
                 <scope>test</scope>
             </dependency>
 
+            <dependency>
```
Comment (Hisoka-X) [inline pom.xml:518]: Hi, I didn't find the reason about add this dependency

File: seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        this.pwd = pwd;
+        this.baseUrl = baseUrl.endsWith("/") ? baseUrl : baseUrl + "/";
+        this.defaultUrl = this.baseUrl + defaultDatabase;
+    }
+
+    /**
+     * URL has to be without database, like "jdbc:postgresql://localhost:5432/" or
+     * "jdbc:postgresql://localhost:5432" rather than "jdbc:postgresql://localhost:5432/db".
+     */
+    public static void validateJdbcUrl(String url) {
```
Comment (Hisoka-X) [inline seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java]: Maybe can create another constructor, use jdbcurl but no `defaultDatabase` parameter, the defaultDatabase set by jdbcurl. then user can set `defaultDatabase` just use url.

File: seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    public String getPassword() {
+        return pwd;
+    }
+
+    public String getBaseUrl() {
+        return baseUrl;
+    }
+
+    @SuppressWarnings("EmptyBlock")
```
Comment (Hisoka-X) [inline seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java]: add a comment in try statement can avoid `EmptyBolck`

File: pom.xml (hunk 1 above)
Comment (ashulin) [reply pom.xml:518]: This is added in `dependencyManagement`.
I expect the new module to be tested using junit5, using @ParameterizedTest, @EnabledOnOs, etc

File: seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java (hunk 2 above)
Comment (ashulin) [reply seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java]: You have a point there.

File: seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java (hunk 3 above)
Comment (ashulin) [reply seatunnel-connectors/seatunnel-connectors-seatunnel/seatunnel-connector-seatunnel-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/AbstractJdbcCatalog.java]: I'll give it a try.

File: pom.xml (hunk 1 above)
Comment (ashulin) [reply pom.xml:518]: I added some parameterized test cases.

Comment (Hisoka-X) [reply pom.xml:518]: If we use junit5, can we remove older version from pom?

Comment (ashulin) [reply pom.xml:518]: If we remove junit4, we need to modify the code of the old API. Are we going to do this?

Comment (Hisoka-X) [reply pom.xml:518]: @CalvinKirs @ruanwenjun

**Pull request merged**